internal/addon/addon.go

```diff
@@ -1,12 +1,21 @@
 package addon
 
 import (
+	"errors"
+	"fmt"
+
 	"open-cluster-management.io/addon-framework/pkg/agent"
 	"open-cluster-management.io/addon-framework/pkg/utils"
 	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	workapiv1 "open-cluster-management.io/api/work/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	ErrWrongType      = errors.New("probe condition is not satisfied")
+	ErrValueNotProbed = errors.New("value not probed")
```

Comment:
I think both don't need to be exposed outside the package
```suggestion
	errWrongType      = errors.New("probe condition is not satisfied")
	errValueNotProbed = errors.New("value not probed")
```